src/core/util/http_client/httpcli.cc

```diff
@@ -171,7 +175,14 @@ HttpRequest::HttpRequest(
       pollent_(pollent),
       pollset_set_(grpc_pollset_set_create()),
       test_only_generate_response_(std::move(test_only_generate_response)),
-      resolver_(GetDNSResolver()) {
+      use_event_engine_dns_resolver_(IsEventEngineDnsNonClientChannelEnabled()),
+      resolver_(!use_event_engine_dns_resolver_ ? GetDNSResolver() : nullptr),
+      ee_resolver_(use_event_engine_dns_resolver_
+                       ? ChannelArgs::FromC(channel_args_)
+                             .GetObjectRef<EventEngine>()
+                             ->GetDNSResolver(
+                                 EventEngine::DNSResolver::ResolverOptions())
+                       : absl::UnknownError("")) {
```

Comment:
Nit: Worth adding a short string for help debugging later if something goes wrong. I suggest `absl::InternalError("EventEngine DNS is not enabled")`, since there's no valid way this error should ever be seen.